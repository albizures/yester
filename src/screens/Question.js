import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Text, View, Button } from 'react-native'
import QuestionCard from './Home/QuestionCard'

export default class Question extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  onPress = () => {
    this.props.navigation.navigate('Writing')
  }

  render () {
    return (
      <View>
        <Text style={[{textAlign: 'center', marginTop: 40}]}>QUESTION</Text>
        <Button title='to Writing' onPress={this.onPress} />
        <View style={{flex: 1, flexDirection: 'row', justifyContent: 'center'}}>
          <QuestionCard text={'Question Card'} />
        </View>
      </View>
    )
  }
}
