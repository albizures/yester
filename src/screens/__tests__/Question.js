import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import Question from '../Question'

const navigation = { navigate: jest.fn() }

describe('src/screens/ConfirmEmail.js', () => {
  it('should render', () => {
    const tree = renderer.create(<Question navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button to writing is pressed', () => {
    it('should navigate to Writing screen', () => {
      const shallowQuestion = shallow(<Question navigation={navigation} />)
      const shallowTouchableHighlights = shallowQuestion.find(Button)
      const shallowButtonToHome = shallowTouchableHighlights.at(0)

      shallowButtonToHome.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.navigate).toHaveBeenCalledWith('Writing')
    })
  })
})
