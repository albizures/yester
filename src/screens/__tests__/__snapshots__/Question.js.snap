// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/ConfirmEmail.js should render 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "marginTop": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    QUESTION
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        to Writing
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "#fff",
          "borderRadius": 30,
          "borderWidth": 0.5,
          "elevation": 8,
          "flex": 1,
          "height": 520,
          "marginBottom": 10,
          "marginTop": 10,
          "shadowColor": "#D5D3DD",
          "shadowOffset": Object {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 15,
          "width": 340,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 30,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-evenly",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "height": 50,
              "justifyContent": "center",
              "marginLeft": 30,
              "width": 220,
            }
          }
        >
          <Text
            numberOfLines={2}
            style={
              Array [
                Object {
                  "color": "#313131",
                  "fontFamily": "Karla-Regular",
                  "fontSize": 18,
                  "textAlign": "left",
                },
                Object {
                  "flexWrap": "wrap",
                },
              ]
            }
          >
            Question Card
          </Text>
        </View>
        <Image
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/button-plus.png",
            }
          }
          style={
            Object {
              "height": 50,
              "marginRight": 17,
              "width": 50,
            }
          }
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
            },
            Object {
              "backgroundColor": "#4241B3",
              "borderColor": "#4241B3",
            },
            Object {
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Primary
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#4241B3",
            },
            Object {
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4241B3",
                },
              ]
            }
          >
            Secundary
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
